### PR TITLE
Update gir-files

### PR DESCRIPTION
**Pull request by sdroege**
This causes some API changes... but before these were just wrong :upside_down_face:

### CODE REVIEW

File: gio/src/auto/file.rs [hunk 1]
```diff
@@ -675,7 +675,7 @@ pub trait FileExt: 'static {
 
     #[doc(alias = "g_file_resolve_relative_path")]
     #[must_use]
-    fn resolve_relative_path(&self, relative_path: impl AsRef<std::path::Path>) -> File;
+    fn resolve_relative_path(&self, relative_path: impl AsRef<std::path::Path>) -> Option<File>;
```
Comment (sdroege) [inline gio/src/auto/file.rs:678]: This

File: glib/src/auto/functions.rs [hunk 2]
```diff
@@ -427,7 +427,7 @@ pub fn user_special_dir(directory: UserDirectory) -> Option<std::path::PathBuf>
 #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_72")))]
 #[doc(alias = "g_get_user_state_dir")]
 #[doc(alias = "get_user_state_dir")]
-pub fn user_state_dir() -> crate::GString {
+pub fn user_state_dir() -> std::path::PathBuf {
```
Comment (sdroege) [inline glib/src/auto/functions.rs:430]: This

File: gio/src/auto/app_launch_context.rs [hunk 3]
```diff
@@ -65,7 +65,7 @@ pub trait AppLaunchContextExt: 'static {
     #[cfg(any(feature = "v2_72", feature = "dox"))]
     #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_72")))]
     #[doc(alias = "launch-started")]
-    fn connect_launch_started<F: Fn(&Self, &AppInfo, &glib::Variant) + 'static>(
+    fn connect_launch_started<F: Fn(&Self, &AppInfo, Option<&glib::Variant>) + 'static>(
```
Comment (sdroege) [inline gio/src/auto/app_launch_context.rs:68]: This

**Comment by sdroege** [comment]
Maybe should be backported to 0.15 but it makes me unhappy that it changes API. OTOH using that API would've caused random panics...

File: gio/src/auto/tls_certificate.rs [hunk 4]
```diff
@@ -115,6 +138,31 @@ impl TlsCertificate {
         }
     }
 
+    #[cfg(any(feature = "v2_72", feature = "dox"))]
+    #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_72")))]
+    #[doc(alias = "g_tls_certificate_new_from_pkcs12")]
+    #[doc(alias = "new_from_pkcs12")]
+    pub fn from_pkcs12(
+        data: u8,
+        length: usize,
```
Comment (sdroege) [inline gio/src/auto/tls_certificate.rs:147]: Missing array length annotation

Comment (sdroege) [reply gio/src/auto/tls_certificate.rs:147]: https://gitlab.gnome.org/GNOME/glib/-/merge_requests/2453

Comment (sdroege) [reply gio/src/auto/tls_certificate.rs:147]: Once that is merged I'll get `gobject-introspection` updated and then we can update the gir files another time.

Comment (bilelmoussaoui) [reply gio/src/auto/tls_certificate.rs:147]: At least from now on we can notice such breakages as early as possible... Hoping to see g-i as part of glib in the future to no longer have to deal with this kind of issues.

Comment (sdroege) [reply gio/src/auto/tls_certificate.rs:147]: https://gitlab.gnome.org/GNOME/gobject-introspection/-/merge_requests/311

**Comment by sdroege** [comment]
@bilelmoussaoui @bilelmoussaoui Opinions?

**Comment by bilelmoussaoui** [comment]
Oups completely forgot about this

**Pull request merged**